tofu/geom/utils.py

```diff
@@ -795,20 +807,36 @@ def _create_config_testcase(config=None, returnas='object',
     for cc in lcls:
         for ss in dconfig[config][cc]:
             ff = [f for f in lf
-                  if all([s in f for s in [cc,Exp,ss]])]
-            if not len(ff) == 1:
-                msg = "No / several matching files\n"
+                  if all([s in f for s in [cc, Exp, ss]])]
+            if len(ff) == 0:
+                msg = "No matching files\n"
                 msg += "  Folder: %s\n"%path
                 msg += "    Criteria: [%s, %s]\n"%(cc,ss)
                 msg += "    Matching: "+"\n              ".join(ff)
                 raise Exception(msg)
-            pfe = os.path.join(path,ff[0])
-            obj = eval('_core.'+cc).from_txt(pfe, Name=ss, Type='Tor',
-                                             Exp=dconfig[config]['Exp'],
-                                             out=returnas)
-            if returnas not in ['object', object]:
-                obj = ((ss,{'Poly':obj[0], 'pos':obj[1], 'extent':obj[2]}),)
-            lS.append(obj)
+            elif len(ff) > 1:
+                # More demanding criterion
+                ssbis, Expbis = '_'+ss+'.txt', '_Exp'+Exp+'_'
```

Comment:
what are ss files ?

Reply:
In this loop, cc is the class str ('PFC', 'Ves'...) and ss is the name ('').
The name of a .txt file containing the polygon of a Struct is formatted as

TFG_class_ExpTokamak_Name.txt

For example:

TFG_PFC_ExpAUG_TPLT1.txt

This file contains the polygon for a PFC of experiment ASDEX Upgrade (AUG) which is called TPLT1.

Reply:
In this method we try to identify a unique txt fle corresponding to a given Struct (by class, Exp, name).
A first try is done with simple criteria which are sufficient 90% of the time.
But in some case multiple matches are found, because for example the names of several Struct share a common part, for example:

TFG_PFC_ExpAUG_D3BG1.txt
TFG_PFC_ExpAUG_D3BG10.txt

These are two different Struct. But the name of the first one is fully enclosed in the name of the second.
So if we look for all files containing the str 'D3BG1', we'll find these two solutions.
In that case, we search again with more strict criteria to identify the unique file.